tests/integration/integration-test.py

```diff
@@ -62,6 +64,191 @@ def test_get_notes_path(self, endpoint='/notes'):
             schema = utils.get_resource_schema(self, 'Error')
             utils.check_schema(self, response, schema)
 
+    def get_response(self, params):
+        response = utils.make_request(self, '/notes', 200, params=params)
+        schema = utils.get_resource_schema(self, 'NoteResource')
+        utils.check_schema(self, response, schema)
+        return response
+
+    # Test case: GET /notes?studentId
+    def test_get_notes_query_by_student_id(self, endpoint='/notes'):
+        for student_id in self.test_cases['valid_student_id']:
+            params = {'studentId': student_id}
+            response = self.get_response(params)
+
+            creator_ids = set()
+            context_types = set()
+            notes = set()
+            sources = set()
+            # Validating the studentId requested is the same studentId received
+            for resource in response.json()['data']:
+                attributes = resource['attributes']
+                actual_student_id = attributes['studentId']
+                self.assertEqual(actual_student_id, student_id)
+
+                note_id = resource['id']
+                match_student_id = re.match(r'^(\d{9})-.+', note_id)
+                if match_student_id:
+                    self.assertEqual(student_id, match_student_id.group(1))
+                else:
+                    self.fail('Note id don\'t include a 9 digit student id')
+
+                creator_ids.add(attributes['creatorId'])
+                context_types.add(attributes['context']['contextType'])
+                notes.add(attributes['note'])
+                sources.add(attributes['source'])
+
+            if len(response.json()['data']) > 1:
+                if len(creator_ids) > 1:
+                    self.query_creator_id(student_id, creator_ids)
+                if len(sources) > 1:
+                    self.query_source(student_id, sources)
+                if len(context_types) > 1:
+                    self.query_context_type(student_id, context_types)
```

Comment:
in a previous comment @tsoliangwu0130 mentioned that if there is only one resource in the response then its not worth it to test. https://github.com/osu-mist/notes-api/pull/18#discussion_r276024250

Reply:
It looks like this was just in reference to testing sort keys, which makes sense since the sorted order of the response couldn't be tested with a singleton array. I don't think this applies to the other fields though.